### ISSUE TITLE
Text with units is mutated by `remove()`

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#967]
This is a repeat of the second issue described in closed issue #960 

The following 2 pieces of code produce 2 different outputs, but they should be the same output?

```javascript
const nlp = require('compromise')
const txt = `Remove me
500mg of paracetamol`

const doc = nlp(txt)
const matches = doc.match('Remove me')
matches.forEach((m) => doc.remove(m))
console.log(doc.text())
```

Output:
❌ `500 mg of paracetamol` - `500mg` is mutated to `500 mg`

```javascript
const nlp = require('compromise')
const txt = `500mg of paracetamol`

const doc = nlp(txt)
console.log(doc.text())
```

✅ `500mg of paracetamol` - no mutation.

Is this a bug or by design? If the latter, can this behaviour be disabled via any configuration parameters? This unexpected mutation only seems to happen on Units and nothing else. It seems to be related to change #919 / Compromise `v14.4.0`

Thanks!

**Comment by spencermountain** [comment]
fixed in `14.6.0`

**Comment by thegoatherder** [comment]
Thanks, @spencermountain! 🙏
